agent/structs/federation_state.go

```diff
@@ -0,0 +1,107 @@
+package structs
+
+import (
+	"sort"
+	"time"
+)
+
+type FederationStateOp string
+
+const (
+	FederationStateUpsert FederationStateOp = "upsert"
+	FederationStateDelete FederationStateOp = "delete"
+)
+
+type FederationStateRequest struct {
```

Comment:
Doc comments for context on these structs would be nice for later readers - we're pretty OK at doing that throughout structs package generally.

Reply:
done